exasol_advanced_analytics_framework/interface/create_event_handler_udf.py

```diff
@@ -1,44 +1,93 @@
+from exasol_bucketfs_utils_python.bucketfs_location import BucketFSLocation
+from exasol_advanced_analytics_framework.event_handler.event_handler_base \
+    import EventHandlerBase
+from exasol_advanced_analytics_framework.event_handler.event_handler_context \
+    import EventHandlerContext
+from exasol_advanced_analytics_framework.event_handler.event_handler_result \
+    import EventHandlerResult
+from exasol_advanced_analytics_framework.event_handler.event_handler_state \
+    import EventHandlerState
+from exasol_advanced_analytics_framework.utils.udf_context_wrapper import \
+    UDFContextWrapper
 
 
 class CreateEventHandlerUDF:
+    STATE_DIRECTORY = "aaf_state"
+
     def __init__(self, exa):
         self.exa = exa
 
     def run(self, ctx) -> None:
+        # get and set method parameters
         parameters = ctx.parameters
         bucketfs_connection = ctx.bucketfs_connection
+        bucketfs_location = BucketFSFactory().create_bucketfs_location(
+            url=bucketfs_connection.address,
+            user=bucketfs_connection.user,
+            pwd=bucketfs_connection.password,
+            base_path=parameters["base"]
+        )
+        bucketfs_path = PurePosixPath(
+            self.STATE_DIRECTORY, f"{parameters['bucket_file_path']}.pkl")
+
+        # load the latest (create if not) event handler state object
+        latest_state = self._load_latest_state(
+            parameters, bucketfs_location, bucketfs_path)
+        event_handler_context: EventHandlerContext = latest_state.context
+        event_handler: EventHandlerBase = latest_state.event_handler
 
-        event__handler_context = EventHandlerContext(bucketfs_connection)
-        event_handler_obj = self._get_event_handler_obj(
-            parameters,  event__handler_context)
-        result: EventHandlerResult = event_handler_obj.handle_event()  # TODO ?
-        self._save_event_handler_obj(event_handler_obj, event__handler_context)
+        # call the user code
+        udf_context = self._create_udf_context_wrapper(ctx)
+        result: EventHandlerResult = event_handler.handle_event(
+            udf_context, event_handler_context)  # TODO does user change state?
 
+        # update and save (overwrite)the current state
+        current_state = EventHandlerState(event_handler_context, event_handler)
+        self._save_current_state(
+            bucketfs_location, bucketfs_path, current_state)
+
+        # return queries
         ctx.emit(result.return_query)
         ctx.emit(result.status)
         for query in result.query_list:
             ctx.emit(query)
 
-    def _get_event_handler_obj(
-            self, parameters, event__handler_context) -> EventHandlerBase:
+    def _load_latest_state(
+            self,
+            parameters: dict,
+            bucketfs_location: BucketFSLocation,
+            bucketfs_path: PurePosixPath) -> EventHandlerState:
+
+        event_handler_state = bucketfs_location.\
```

Comment:
This will throw an exception if the file is not available. I suggest, you check if the file is actually available. Maybe, it is also better, to explicitly inform the UDF about which state to load or if it gets initialized.

Reply:
not sure about the way of explicitly informing UDF,  since we have to manipulate sql query where we invoke  in event loop (saying that it is the first call). did you mean that too?

I guess we can't check whether the file exists with the bucketfs_location api. Exception handling is not a good way.

Reply:
We might need to extend the bucketfs utils for that. Yes, I mean that event loop could inform the UDF that this is the first call. You could use different mechanism for that. Either you UDF parameter or use a connection object. For example, you could add a column to the UDF input, which contains either the location of the checkpoint or the config.